cmd/abapEnvironmentCheckoutBranch.go

```diff
@@ -0,0 +1,138 @@
+		CookieJar:          cookieJar,
+		Username:           connectionDetails.User,
+		Password:           connectionDetails.Password,
+	}
+	client.SetOptions(clientOptions)
+	pollIntervall := 10 * time.Second
+
+	log.Entry().Infof("Starting to switch branch to branch '%v' on repository '%v'", options.BranchName, options.RepositoryName)
+	log.Entry().Info("--------------------------------")
+	log.Entry().Info("Start checkout branch: " + options.BranchName)
+	log.Entry().Info("--------------------------------")
+
+	// Triggering the Checkout of the repository into the ABAP Environment system
+	uriConnectionDetails, errorTriggerCheckout := triggerCheckout(options.RepositoryName, options.BranchName, connectionDetails, &client)
+	if errorTriggerCheckout != nil {
+		log.Entry().WithError(errorTriggerCheckout).Fatal("Checkout of " + options.BranchName + " for software component " + options.RepositoryName + " failed on the ABAP System")
+	}
+
+	// Polling the status of the repository import on the ABAP Environment system
+	status, errorPollEntity := abaputils.PollEntity(options.RepositoryName, uriConnectionDetails, &client, pollIntervall)
+	if errorPollEntity != nil {
+		log.Entry().WithError(errorPollEntity).Fatal("Status of checkout action on repository" + options.RepositoryName + " failed on the ABAP System")
+	}
+	if status == "E" {
+		log.Entry().Fatal("Checkout of branch " + options.BranchName + " failed on the ABAP System")
+	}
+
+	log.Entry().Info("--------------------------------")
+	log.Entry().Infof("Checkout of branch %v on repository %v was successful", options.BranchName, options.RepositoryName)
+	log.Entry().Info("--------------------------------")
+	return nil
+}
+
+func triggerCheckout(repositoryName string, branchName string, checkoutConnectionDetails abaputils.ConnectionDetailsHTTP, client piperhttp.Sender) (abaputils.ConnectionDetailsHTTP, error) {
+
+	uriConnectionDetails := checkoutConnectionDetails
+	uriConnectionDetails.URL = ""
+	checkoutConnectionDetails.XCsrfToken = "fetch"
+
+	// Loging into the ABAP System - getting the x-csrf-token and cookies
+	resp, err := abaputils.GetHTTPResponse("HEAD", checkoutConnectionDetails, nil, client)
+	if err != nil {
+		err = abaputils.HandleHTTPError(resp, err, "Authentication on the ABAP system failed", checkoutConnectionDetails)
+		return uriConnectionDetails, err
+	}
+	defer resp.Body.Close()
+	log.Entry().WithField("StatusCode", resp.Status).WithField("ABAP Endpoint", checkoutConnectionDetails.URL).Info("Authentication on the ABAP system was successful")
+	uriConnectionDetails.XCsrfToken = resp.Header.Get("X-Csrf-Token")
+	checkoutConnectionDetails.XCsrfToken = uriConnectionDetails.XCsrfToken
+
+	// Initial checks
+	if repositoryName == "" {
+		return uriConnectionDetails, errors.New("An empty string was passed for the parameter 'repositoryName'")
+	}
+	if branchName == "" {
+		return uriConnectionDetails, errors.New("An empty string was passed for the parameter 'branchName'")
+	}
```

Comment:
Are they required? If the property has the "mandatory" flag (in the yaml file), this should be checked by the "framework"

Reply:
Fixed